packages/graphql/src/types/app.ts

```diff
@@ -152,10 +175,49 @@ export type AllQuestionInstanceTypeData =
 
 declare global {
   namespace PrismaJson {
-    type PrismaAvatarSettings = AvatarSettings
     type PrismaQuestionResponse = QuestionResponse
-    type PrismaQuestionOptions = QuestionOptions
+    type PrismaElementOptions = ElementOptions
     type PrismaQuestionResults = QuestionResults
-    type PrismaQuestionData = AllQuestionTypeData
+    type PrismaElementData = AllElementTypeData
+  }
+}
+// #endregion
+
+// ----- ELEMENT STACKS -----
+// #region
+export type LiveQuizStackOptions = {
+  timeLimit?: number
+  expiresAt?: Date
+  randomSelection?: number
+  // TODO: by moving it here, we lose the default value, so ensure it is set in backend code correctly
+  execution: number
+  // TODO: rename the enum for consistency
+  status: SessionBlockStatus
+}
+
+export type MicrolearningStackOptions = {}
+
+export type PracticeQuizStackOptions = {}
+
+export type ElementStackOptions =
+  | LiveQuizStackOptions
+  | MicrolearningStackOptions
+  | PracticeQuizStackOptions
+
+interface IElementStackOptions<
+  Type extends ElementStackType,
+  Options extends ElementStackOptions | null
+> {}
+
+export type AllElementStackOptions =
+  | IElementStackOptions<'LIVE_QUIZ', LiveQuizStackOptions>
```

Comment:
Do we have to use Strings as types here and cannot use more general types like ElementStackType.LIVE_QUIZ?

Reply:
same as previously. types cannot be used here, as TS expects values. in some way we tell it "if the runtime value is LIVE_QUIZ, the corresponding type will be LiveQuizStackOptions". this will allow TS to narrow down to the type when we have an if statement checking for the type